Makefile

```diff
@@ -93,6 +94,15 @@ configure-installer-tests-cluster:
 test:
 	go test -race $(PKGS)
 
+# Run clean test template spec parallelly
+.PHONY: clean-test
+clean-test:
+ifeq ($(SPEC_EXEC_METHOD),series)
+	go test -v github.com/openshift/odo/tests/template --ginkgo.focus="Example of a clean test" -ginkgo.slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -ginkgo.v -timeout $(TIMEOUT)
+else
+	ginkgo -v -nodes=4 -focus="Example of a clean test" slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -randomizeAllSpecs  tests/template/ -timeout $(TIMEOUT)
```

Comment:
@kadel 
```
Usage of ginkgo:
  [...]
  -nodes int
        The number of parallel test nodes to run (default 1)
  [...]
```
As per the ginkgo uses and my observation node=1 runs the test one by one (sequential run). But if the node is increased then the no os spec are distributed to run in parallel. 

For example : 
node=4 runs four spec in parallel on different node at a time and rest of the specs if any will be in queue.

> The number of nodes should be configurable using a variable.

Yes, it can be doable. Then there would be more complexity to justify node limit. For example i can run test smoothly on 4 node easily but if i set node 10 then my local cluster become dead slow. 

That is why i fixed the node=4 without exposing it through env variable.

Reply:
> > As per @kadel 's comments, I also request that we move this code out of the Makefile as it serves no purpose (it's pretty much running just template code?). Instead, documentation and scripts would make sense, even if it's a hack/gen-e2e-test.sh script.
> 
> Yes, it serves the purpose of how to write the clean test and execute it though it's not part of CI. But your suggestion is also worth having.

I think that how to execute it should be covered in the comments or docs.

Reply:
Sorry, I'm still missing a point why we can't just use a variable for `-nodes` and remove `SPEC_EXEC_METHOD`.

What I meant was that we can have `TEST_NODES` or `PARALELL_NODES` or something like this. If set to `1` then it would mean that tests will be run in series (the same result as setting `SPEC_EXEC_METHOD` to `series` and if `TEST_NODES>1` then it would be parallel run with a given number of nodes.

Reply:
> Sorry, I'm still missing a point why we can't just use a variable for `-nodes` and remove `SPEC_EXEC_METHOD`.
> 
> What I meant was that we can have `TEST_NODES` or `PARALELL_NODES` or something like this. If set to `1` then it would mean that tests will be run in series (the same result as setting `SPEC_EXEC_METHOD` to `series` and if `TEST_NODES>1` then it would be parallel run with a given number of nodes.

I got the context. Looks promising. Will try it out

Reply:
Done